### PR TITLE
Pipeline for keyvault secrets 

### LINKED ISSUE #2607: Setup pipeline for secrets
cy.yml 
testresources.json

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

closes #2607

### CODE REVIEW

**Comment by gearama** [comment]
@danieljurek pipeline file changes

File: sdk/keyvault/test-resources.json [hunk 1]
```diff
@@ -1,223 +1,243 @@
 {
-    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
```
Comment (gearama) [inline sdk/keyvault/test-resources.json:2]: All the red part is due to vs reformatting the code, the useful changes contain the word secret

**Review by danieljurek — APPROVED** [review]
Changes LGTM. YAML changes look correct and the ARM template deploys correctly. 👍

File: sdk/keyvault/test-resources.json (hunk 1 above)
Comment (vhvb1989) [reply sdk/keyvault/test-resources.json:2]: I am a little worry about the re-formatting.  Do we know if it won't be re-formatted later from VSCode or some other tool?  
Do we have some format config for json?

**Review by vhvb1989 — APPROVED** [review]
Maybe share what tool/version you used to format the json file. Then we can try to see if it can become our standard for json files.
Is it following what other languages are using for json format? (just curious).

We use clang format for the code, but I am not sure if it would cover json formats.

**Comment by vhvb1989** [comment]
It's alive!

image

**Pull request merged**